packages/react-components/react-list-preview/src/components/ListItem/useListItem.tsx

```diff
@@ -45,46 +66,119 @@ export const useListItem_unstable = (
   const parentRenderedAs = useListContext_unstable(ctx => ctx.as);
   const renderedAs = props.as || DEFAULT_ROOT_EL_TYPE;
 
+  const rootRef = React.useRef<HTMLLIElement | HTMLDivElement>(null);
+
   validateProperElementTypes(parentRenderedAs, renderedAs);
 
-  const focusableGroupAttrs = useFocusableGroup({ tabBehavior: 'limited-trap-focus' });
+  validateNavigableWhenOnClickPresent(focusableItems, onClick);
+
+  const { findFirstFocusable, findPrevFocusable } = useFocusFinders();
+
+  const baseIndicatorStyles = useIndicatorStyle();
+
+  const focusableGroupAttrs = useFocusableGroup({ ignoreDefaultKeydown: { Enter: true } });
+
+  const handleClick: React.MouseEventHandler<HTMLLIElement & HTMLDivElement> = useEventCallback(e => {
+    onClick?.(e);
+
+    if (!isSelectionEnabled || e.defaultPrevented) {
+      return;
+    }
+
+    toggleItem?.(e, value);
+  });
 
   const handleKeyDown: React.KeyboardEventHandler<HTMLLIElement & HTMLDivElement> = useEventCallback(e => {
     onKeyDown?.(e);
 
-    // Compare targets to make sure this only triggers when the event is fired on the list item
-    // and not on a button inside
-    if (!isSelectionEnabled || e.defaultPrevented || e.target !== e.currentTarget) {
+    // Return early if prevented default
+    if (e.defaultPrevented) {
+      return;
+    }
+
+    // If the list items themselves are focusable and the event is fired from an element inside the list item
+    if (focusableItems && e.target !== e.currentTarget) {
+      // If it's one of the Arrows defined, jump out of the list item to focus on the ListItem itself
+      // The ArrowLeft will only trigger if the target element is the leftmost, otherwise the
+      // arrowNavigationAttributes handles it and prevents it from bubbling here.
+      if (['ArrowLeft', 'ArrowUp', 'ArrowDown'].includes(e.key)) {
+        // Prevents scrolling for ArrowUp/ArowDown
+        e.preventDefault();
+        e.target.dispatchEvent(
+          new KeyboardEvent('keydown', {
+            key: Escape,
+            keyCode: keyCodes.Escape,
+          }),
+        );
+      }
+    }
+
+    // Now return early if the event is not fired from the list item itself
+    // as the following code handles the events specifically coming from the list item
+    if (e.target !== e.currentTarget) {
       return;
     }
 
-    if (e.key === Space) {
+    // Handle selection for enter and space
+    if ((isSelectionEnabled || typeof props.onClick === 'function') && [Space, Enter].includes(e.key)) {
```

Comment:
Move array to module scope:

```suggestion
    if ((isSelectionEnabled || typeof props.onClick === 'function') && SELECTION_KEYS.includes(e.key)) {
```